org.mixedrealitytoolkit.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs

```diff
@@ -609,6 +634,49 @@ private void InstantiateManipulationLogic()
             };
         }
 
+        private void EnsureDefaultManipulationSpecializations()
+        {
+            if (!addDefaultManipulationSpecializations)
+            {
+                return;
+            }
+
+            // Ensure the default set of manipulation specializations is present.
+            gameObject.EnsureComponent<SocketManipulationSpecialization>();
+        }
+
+        private void UpdateSelectManipulationSpecialization()
+        {
+            var previousSpecialization = activeSpecialization;
+            activeSpecialization = null;
+
+            foreach (var specialization in GetComponents<IManipulationSpecialization>())
```

Comment:
Do we really need multiple specializations?   Seems like your scenario only needs one.

Let's have the developer explicitly set the specialization logic, instead of requiring querying GetComponents.  This query will impact performance even for non-specialized case.